### PR TITLE
https://unit.atlassian.net/browse/NP-45530 qfix AND

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search2/model/OpenSearchQuery.java [hunk 1]
```diff
@@ -127,11 +125,7 @@ public boolean hasNoSearchValue() {
     public void setSearchFieldValue(K key, String value) {
         if (nonNull(value)) {
             var decodedValue = key.valueEncoding() != ValueEncoding.NONE ? decodeUTF(value) : value;
-            if (key.fieldType() == ParamKind.STRING) {
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/model/OpenSearchQuery.java:130]: query values should not be modified, only when values are transformed into query

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java [hunk 2]
```diff
@@ -146,9 +147,10 @@ private OpenSearchSwsResponse handleResponse(HttpResponse<String> response) {
     }
 
     private MultiMatchQueryBuilder multiMatchQuery(ResourceAwsQuery query) {
-        var field = query.getValue(FIELDS).toString();
+        var fields = extractFields(query.getValue(FIELDS).toString());
+        var value = escapeSearchString(query.getValue(SEARCH_ALL).toString());
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:151]: query values should not be modified, only when values are transformed into query

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java [hunk 3]
```diff
@@ -158,13 +160,17 @@ private BoolQueryBuilder boolQuery(ResourceAwsQuery query) {
             .forEach((key, value) -> {
                 var searchFields = key.searchFields().toArray(String[]::new);
                 if (hasMultipleFields(searchFields)) {
-                    bq.must(QueryBuilders.multiMatchQuery(value, searchFields));
+                    bq.must(QueryBuilders
+                                .multiMatchQuery(escapeSearchString(value), searchFields)
+                                .operator(Operator.AND));
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:165]: q fix AND

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java [hunk 4]
```diff
@@ -158,13 +160,17 @@ private BoolQueryBuilder boolQuery(ResourceAwsQuery query) {
...
                 } else {
                     var searchField = searchFields[0];
                     if (isRangeQuery(key)) {
                         bq.must(rangeQuery(key.searchOperator(), searchField, value));
                     } else {
-                        bq.must(QueryBuilders.matchQuery(searchField, value));
+                        bq.must(QueryBuilders
+                                    .matchQuery(searchField, escapeSearchString(value))
+                                    .operator(Operator.AND));
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:173]: q fix AND

**Review by brinxmat — APPROVED** [review]

**Pull request merged**